pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,290 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+import numpy as np
+import re
+
+from pyfr.inifile import _ensure_float
+from pyfr.libs.ascent import Ascent
+from pyfr.libs.conduit import ConduitNode
+from pyfr.mpiutil import get_comm_rank_root
+from pyfr.nputil import npeval
+from pyfr.plugins.base import BasePlugin
+from pyfr.shapes import BaseShape
+from pyfr.util import file_index, file_path, subclass_where
+from pyfr.writers.vtk import BaseShapeSubDiv
+
+
+class AscentPlugin(BasePlugin):
+    name = 'ascent'
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    def __init__(self, intg, cfgsect, suffix=None):
+        super().__init__(intg, cfgsect, suffix)
+
+        comm, rank, root = get_comm_rank_root()
+        self.nsteps = self.cfg.getint(cfgsect, 'nsteps')
+
+        # Underlying system
+        system = intg.system
+        self._ele_map = system.ele_map
```

Comment:
Do not hand on to the elemap; it is a big memory hog.  Pull about the bits you need in the constructor.